server/src/main/java/io/druid/segment/realtime/firehose/WikipediaIrcDecoder.java

```diff
@@ -90,35 +90,59 @@ public WikipediaIrcDecoder(
     }
     this.namespaces = namespaces;
     this.geoIpDatabase = geoIpDatabase;
+    this.geoLookup = openGeoDB(geoIpDatabase);
+  }
 
-    File geoDb;
-    if (geoIpDatabase != null) {
-      geoDb = new File(geoIpDatabase);
-    } else {
+  private DatabaseReader openGeoDB(String configDbPath) {
+    RuntimeException lastEx = null;
+    for (int i=0; i<2; i++) {
```

Comment:
It is a retry for when the config does not have a geo db file path specified (ie, we're downloading it into a temp space), but opening that file results in an exception. If it opens fine, it doesn't retry. If it throws an exception, but it was specified in the config, it will also not retry. 

In my case, the decoder was failing in the constructor because the file was only partially downloaded (process was killed mid-download), so the server was never coming up. The thought was that if it could detect this and just try to re-download the db, it could recover without user intervention (where the user has to find and delete the file). So, this retry will delete it and then attempt to download and open it again. However, the retry may be not be necessary with the change that has the download writing to a .tmp extension until fully downloaded.